CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required (VERSION 3.2)
+project (optee_examples C)
+
+# https://cmake.org/Wiki/CMake_Useful_Variables
+set (CMAKE_TOOLCHAIN_FILE CMakeToolchain.txt)
+
+include(GNUInstallDirs)
+
+add_compile_options (-Wall)
+#add_compile_options (
+#	-Wall -Wbad-function-cast -Wcast-align
+#	-Werror-implicit-function-declaration -Wextra
+#	-Wfloat-equal -Wformat-nonliteral -Wformat-security
+#	-Wformat=2 -Winit-self -Wmissing-declarations
+#	-Wmissing-format-attribute -Wmissing-include-dirs
+#	-Wmissing-noreturn -Wmissing-prototypes -Wnested-externs
+#	-Wpointer-arith -Wshadow -Wstrict-prototypes
+#	-Wswitch-default -Wunsafe-loop-optimizations
+#	-Wwrite-strings -Werror -fPIC
+#)
+
+find_program(CCACHE_FOUND ccache)
+if(CCACHE_FOUND)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
+    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
+endif(CCACHE_FOUND)
+
+add_subdirectory (aes)
+add_subdirectory (hotp)
+add_subdirectory (random)
+add_subdirectory (hello_world)
```

Comment:
How about building the list dynamically like in https://github.com/linaro-swg/optee_examples/pull/13/files? For consistency at least.

Reply:
I'm not sure if that is even possible to have something like that in a CMakeLists.txt file. If it is possible it would be worth double check that it doesn't affect CMake in a negative way, i.e. tricking CMake into doing unnecessary (re-)builds etc.

Reply:
This (untested):
```cmake
file(GLOB dirs *)
foreach(dir ${dirs})
  if(EXISTS ${dir}/CMakeLists.txt)
    add_subdirectory(${dir})
  endif()
endforeach()
```

Reply:
You realize that you are replacing 4 lines with 6 lines and at the same time making it a bit more magic? :) ... but I do see the point if/when the number of examples grows.

Reply:
We accepted the idea of enumerating apps in https://github.com/linaro-swg/optee_examples/pull/13 so I don't see why CMake should behave differently...

Reply:
That's a oneliner, but once again, I leave it up to you guys, I'm just giving my feedback.